src/safeposix/syscalls/net_calls.rs

```diff
@@ -296,35 +353,47 @@ impl Cage {
         0
     }
 
+    //bind_syscall implementation in the case that the socket's domain is INET
     fn bind_inner_socket_inet(
         &self,
         sockhandle: &mut SocketHandle,
         newsockaddr: &mut interface::GenSockaddr,
         prereserved: bool,
     ) -> i32 {
-        // INET Sockets
+        //INET Sockets
+        //rebind ability is set to true if the SO_REUSEPORT bit is set in 
+        //the socket handle options
         let intent_to_rebind = sockhandle.socket_options & (1 << SO_REUSEPORT) != 0;
+        //Create a socket and insert it into the innersocket in sockhandle
         Self::force_innersocket(sockhandle);
 
+        //If socket address is preserved, set the local port to the reserved port
+        //Otherwise, set the local port to a new value
         let newlocalport = if prereserved {
             newsockaddr.port()
         } else {
+            //Reserve a new local port num
             let localout = NET_METADATA._reserve_localport(
                 newsockaddr.addr(),
                 newsockaddr.port(),
                 sockhandle.protocol,
                 sockhandle.domain,
                 intent_to_rebind,
             );
+            //If reserving a local port causes an error, return the error num
```

Comment:
You could possibly do the next few lines as a match statement instead.  It would make it clearer that your last unwrap is safe...

I'll note, it is slightly weird that the code here really just wants the inner result regardless of if it is an err or not...